experimental/devices/ccs811/ccs811.go

```diff
@@ -0,0 +1,361 @@
+}
+
+// ReadRawData provides current and voltage on the sensor
+// current in uA
+// voltage from 0-1023, where 1023 = 1.65V
+func (d *Dev) ReadRawData() (current int, voltage int, err error) {
+	r := make([]byte, 2)
+	if err = d.c.Tx([]byte{rawDataReg}, r); err != nil {
+		return 0, 0, err
+	}
+	current, voltage = valuesFromRawData(r)
+	return
+}
+
+// SetEnvironmentData allows to provide temperature and humidity so
+// sensor can compensate it's measurement
+func (d *Dev) SetEnvironmentData(temp, humidity float32) error {
+	rawTemp := uint16((temp + 25) * 512)
+	rawHum := uint16(humidity * 512)
+	w := []byte{environmentReg,
+		byte(rawHum >> 8),
+		byte(rawHum),
+		byte(rawTemp >> 8),
+		byte(rawTemp)}
+
+	if err := d.c.Tx(w, nil); err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetBaseline provides current baseline used by measurement alogrithm
+func (d *Dev) GetBaseline() ([]byte, error) {
+	r := make([]byte, 2)
+	if err := d.c.Tx([]byte{baselineReg}, r); err != nil {
+		return r, err
+	}
+	return r, nil
+}
+
+/*
+SetBaseline sets current baseline for measurement algorithm.
+For mor detail check sensor's specification
+
+Manual Baseline Correction
+There is a mechanism within CCS811 to manually save and restore a previously
+saved baseline value using the BASELINE register. The correct time to save the baseline
+will depend on the customer use-case and application.
+
+• For devices which are powered for >24 hours at a time:
+• During the first 500 hours – save the baseline every
+24-48 hours.
+• After the first 500 hours – save the baseline every 5-7 days.
+
+• For devices which are powered <24 hours at a time:
+• If the device is run in, save the baseline before power down
+• If multiple operating modes are used, a separate baseline should be stored for each
+• The baseline should only be restored when the resistance is stable (typically 20-30 minutes)
+• If changing from a low to high power mode (without spending at least 10 minutes in idle),
+  the sensor resistance should be allowed to settle again before restoring the baseline
+
+Note(s):
+1) If a value is written to the BASELINE register while the sensor is stabilising, the output of the TVOC and eCO2 calculations may be higher than expected.
+2) The baseline must be written after the conditioning period
+*/
+func (d *Dev) SetBaseline(baseline []byte) error {
+	w := []byte{baselineReg, baseline[0], baseline[1]}
+	if err := d.c.Tx(w, nil); err != nil {
+		return err
+	}
+	return nil
+}
+
+// SensorValues represents data read from the sensor.
+// Data are populated based on NeededData parameter.
+type SensorValues struct {
+	ECO2           int
+	VOC            int
+	Status         byte
+	ErrorID        SensorErrorID
+	RawDataCurrent int // current in uA
+	RawDataVoltage int // voltage from 0-1023, where 1023 = 1.65V
+}
+
+// Sense provides data from the sensor.
+// ReadData parameter specifies which data should be read.
+func (d *Dev) Sense(mode NeededData) (*SensorValues, error) {
```

Comment:
make SensorValues a parameter instead of a return value. This way there's no heap allocation on the success case.